src/Producers/DecayChannelProducer.cc

```diff
@@ -1006,7 +996,8 @@ void Run2DecayChannelProducer::Produce(event_type const& event, product_type& pr
 		for (std::vector<KElectron*>::iterator electron = product.m_validElectrons.begin();
 			 electron != product.m_validElectrons.end(); ++electron)
 		{
-			if ((*looseElectron)->p4 == (*electron)->p4)
+			//Use only direction to match loose electrons to the signal electron, since the Pt and mass of signal electrons might be corrected.
+			if (Utility::ApproxEqual((*looseElectron)->p4.Eta(),(*electron)->p4.Eta()) && Utility::ApproxEqual((*looseElectron)->p4.Phi(),(*electron)->p4.Phi()))
```

Comment:
It depends on when you run the corrections producer whether or not this statement is needed. I think we agreed before christmas that the corrections should be applied to all leptons when a corrections producer is run. That is why in the SM configs the any corrections producers are run before any ValidLeptonsProducers are run.

Can this be adopted in MSSM as well? Because then we can go back to comparing the four vectors.